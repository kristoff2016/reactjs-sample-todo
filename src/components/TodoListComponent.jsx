import React from 'react'
import { CCard, CCardBody, CFormGroup, CInputCheckbox, CLabel } from '@coreui/react'
import { TO_DO_STATUS } from 'src/utils/constant'

const TodoListComponent = ({ 
  item,
  onChange
 }) => {
  const checkStatus = (status) => {
    return status === TO_DO_STATUS.COMPLETE ? true : false
  }
  return (
    <CCard style={{
      marginBottom: 10
    }}>
      <CCardBody>
        <CFormGroup variant="custom-checkbox" inline>
          <CInputCheckbox 
            data-testid={item.id}
            onChange={() => onChange(item)}
            custom 
            id={item.id} 
            name='checkbox' 
            value={item.id} 
            checked={checkStatus(item.status)}
          />
          <CLabel variant="custom-checkbox" 
            style={{
             textDecoration: item.status === TO_DO_STATUS.COMPLETE ? 'line-through' : 'none'
            }} 
            htmlFor={item.id}>{item.title}</CLabel>
        </CFormGroup>
      </CCardBody>
    </CCard>
  )
}

export default TodoListComponent