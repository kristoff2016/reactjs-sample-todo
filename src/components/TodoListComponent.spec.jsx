
import { render, fireEvent } from '@testing-library/react';
import React from 'react'
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';

import TodoListComponent from './TodoListComponent'

describe('#TodoListComponent', () => {
  let component = null;
  const initialState = {
    id: 1,
    title: 'sample to do',
    status: 'COMPLETE'
  }
  afterEach(() => {
    jest.clearAllMocks();
  })
  describe('# initialState', () => {
    beforeEach(async () => {
      await act(async () => {
        component = shallow(<TodoListComponent item={initialState} />);
      });
    });
    it('Should render TodoListComponent to toMatchSnapshot', () => {
      const { container } = render(<TodoListComponent item={initialState} />);
      expect(container).toMatchSnapshot();
    })
    it('should render onChange toBeDefined', () => {
      const onChange = jest.fn();
      const { getByTestId } = render(<TodoListComponent item={initialState} onChange={onChange}/>);
      const checkboxId = getByTestId('1');
      expect(checkboxId).toBeDefined();
    });
    it('should render onChange value', () => {
      const onChange = jest.fn()
      const { getByTestId } = render(<TodoListComponent item={initialState} onChange={onChange}/>);
      const checkboxId = getByTestId('1')
      fireEvent.click(checkboxId, initialState)
      expect(onChange).toHaveBeenCalledWith(initialState)
    })
    it('should render onChange true', () => {
      const { getByTestId } = render(<TodoListComponent item={initialState} />);
      const checkboxId = getByTestId('1')
      expect(checkboxId.checked).toBe(true)
    })
  })
})