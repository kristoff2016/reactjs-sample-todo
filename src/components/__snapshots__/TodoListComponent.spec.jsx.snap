// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#TodoListComponent # initialState Should render TodoListComponent to toMatchSnapshot 1`] = `
<div>
  <div
    class="card"
    style="margin-bottom: 10px;"
  >
    <div
      class="card-body"
    >
      <div
        class="custom-control custom-checkbox custom-control-inline"
      >
        <input
          checked=""
          class="custom-control-input"
          data-testid="1"
          id="1"
          name="checkbox"
          type="checkbox"
          value="1"
        />
        <label
          class="custom-control-label"
          for="1"
          style="text-decoration: line-through;"
        >
          sample to do
        </label>
      </div>
    </div>
  </div>
</div>
`;
